.travis.yml

```diff
@@ -39,60 +31,87 @@ env:
     - MYSQL_USER=root
     - PATH="$(git config -f .gitmodules submodule.beaver.path)/bin:$PATH"
     - SKIP_TEST_ON_ERROR=0
-  matrix:
-    - COMPILER_VERSION=4.9 MAKE_TARGET="$MAKE_TARGET:deb" DEPLOY_PACKAGE=true
-    - CXX=clang++ CC=clang COMPILER_VERSION=3.9
-    - DIST_NAME=centos DIST_VERSION=7 MAKE_TARGET=rpm
 
-## Build matrix
-#
-#  Travis expands the environment variables into 6¹ jobs:
-#  All linux based builds are run in docker containers using docker-compose
-#
-#  | os     | compiler    | make targets        |
-#  |--------|-------------|---------------------|
-#  | linux  | clang       | check               |
-#  | linux  | gcc         | check deb           |
-#  | linux  | gcc         | rpm                 |
-#  | osx    | clang 6.1.0 |                     |
-#  | osx    | clang 7.3.0 |                     |
-#  | osx    | clang 8.1.0 |                     |
-#
-#  For osx, the compiler reported is the Apple LLVM version
-#
-#  ¹ Due to excessive build times for osx only XCode 7.3.1 has been enabled.
-matrix:
-  fast_finish: true
-  include:
-    # - os: osx
-    #   osx_image: xcode6.4
-    #   compiler: clang
-    - os: osx
-      compiler: clang
-    # - os: osx
-    #   osx_image: xcode8.3
-    #   compiler: clang
-
-## Install dependencies
+## Build steps for all jobs
 before_install:
   - ./travis_configure.sh before_install
-
-## Prepare build system for tests
 before_script:
   - ./travis_configure.sh before_script
-
-## Run tests, and packaging scripts
 script:
   - ./travis_configure.sh run_tests
 
-## Deploy debian packages to bintray
-##
-## deb package built on ubuntu xenial with gcc
-deploy:
-    provider: script
-    script: beaver bintray upload -D $DIST_VERSION -N build/pkg/deb/*.deb
-    skip_cleanup: true
-    on:
-        tags: true # must be a git tag
-        repo: sociomantic-tsunami/libdrizzle-redux # must be upstream
-        condition: $TRAVIS_OS_NAME = 'linux' && $DIST_NAME = 'ubuntu' && $DIST_VERSION = 'xenial' && $CC = 'gcc' && $MAKE_TARGET =~ 'deb' && $DEPLOY_PACKAGE = 'true'
+## Defines the order of the build stages.
+#
+#  The deployment stage is only done is building semver git tag
+stages:
+  - test
+  - name: deploy
+    if:
+      $TRAVIS_OS_NAME = 'linux' AND
+      env(DIST_NAME) = 'ubuntu' AND
+      env(DIST_VERSION) = 'xenial'
```

Comment:
This can actually be put in the deploy job. I think is more clear to have all in one place.

Reply:
Realised it isn't required since the vars above in `env.global` have exactly the values I check for